clang/test/SemaSYCL/decomposition.cpp

```diff
@@ -53,6 +53,10 @@ struct StructWithPtr {
   int i;
 };
 
+struct Nested {
+typedef StructWithPtr TDStrWithPTR;
```

Comment:
I know the larger reproducer used a typedef and it probably good we have a test with typedef also now but I am curious to know if its required to trigger this crash.